IBAnimatable/Animatable.swift

```diff
@@ -66,7 +66,34 @@ public extension Animatable where Self: UIView {
     }
   }
 
-  public func animate(animation: AnimationType? = nil, completion: AnimatableCompletion? = nil) {
+  public func animate(animation: AnimationType) -> AnimationPromise<Self> {
+    let promise = AnimationPromise(view: self)
+    return promise.then(animation)
+  }
+
+  public func animate(animation: AnimationType? = nil, completion: AnimatableCompletion? = nil) {// here for retro-compatibility
+   // self.animate(animation: animation, promise: nil)
+   // self.animate(animation)
+    self.animate(animation: animation ?? self.animationType).completion(completion)
+  }
+
+  public func delay(delay: Double) -> AnimationPromise<Self> {
```

Comment:
Does it mean we can call `delay` first like `self.blurEffectView.delay(1).animate(animation: .pop(repeatCount: 1))`?